discord-bot/src/interactions/application_commands/matchmaking.rs

```diff
@@ -237,6 +264,153 @@ impl InteractionHandler for MatchmakingCommandHandler {
+                }
+
+                let wins_option_value = options
+                    .get(1)
+                    .ok_or_else(|| anyhow!("could not get wins option"))?
+                    .value
+                    .clone();
+                
+                let loses_option_value = options
+                    .get(2)
+                    .ok_or_else(|| anyhow!("could not get loses option"))?
+                    .value
+                    .clone();
+                
+                let wins;
+                match wins_option_value {
+                    CommandOptionValue::Integer(x) => wins = x as i64,
+                    _ => wins = 0
+                }
+                
+                let loses;
+                match loses_option_value {
+                    CommandOptionValue::Integer(x) => loses = x as i64,
+                    _ => loses = 0
+                }
+
+                // TODO: Refactor all this message stuff to reusable functions with parameters?
+                // TODO: Add an embed with options for the opponent to accept or dispute the score report.
+                // TODO: Format the message so it tags both the User & Opponent.
+                //       And show the wins for both of them in a nice looking way.
+                let guild_settings = self
+                    .utils
+                    .get_guild_settings(
+                        data.command
+                            .guild_id
+                            .ok_or_else(|| anyhow!("command cannot be used in a DM"))?,
+                    )
+                    .await?;
+                let channel;
+                if let Some(cid) = guild_settings.channel_id {
+                    //
+                    // TODO: make sure that the channel actually exists.
+                    //
+                    channel = cid.into_id();
+                } else {
+                    channel = data
+                        .interaction
+                        .channel_id
+                        .ok_or_else(|| anyhow!("command was not run in a channel"))?;
+                }
+                let _followup = self
+                    .utils
+                    .http_client
+                    .interaction(self.utils.application_id)
+                    .create_followup(data.interaction.token.as_str())
+                    .content(format!("Sent a score report in <#{}>", channel).as_str())?
+                    .embeds(&[EmbedBuilder::new()
+                        .title("New score report")
+                        .description(format!(
+                            "{:?} - {:?}",
+                            wins, loses
+                        ))
+                        .validate()?
+                        .build()])?
+                    .flags(MessageFlags::EPHEMERAL)
+                    .exec()
+                    .await?
+                    .model()
+                    .await?;
```

Comment:
I think this is the only part of this that I'd like to change. Scores definitely need to be tracked in the database, but I'm not totally sure how. There needs to be some discussion with how match results are stored. Possibly adding a `MatchResults` model and store some data about the match?